### PR TITLE
move OpenLineage airflow integration to OpenLineage repository

### PR DESCRIPTION
**Pull request by mobuchowski**
Signed-off-by: Maciej Obuchowski <maciej.obuchowski@getindata.com>

### CODE REVIEW

File: integration/airflow/.python-version [hunk 1]
```diff
@@ -0,0 +1 @@
+3.9.2
```
Comment (wslulciuc) [inline integration/airflow/.python-version:1]: Do we need to check in `..python-version`?

File: integration/airflow/README.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+`great_expectations` extractor requires more care than that. For technical reasons, you need to manually provide dataset
+name and namespace for dataset provided to great expectations operator by calling function `marquez_airflow.extractors.great_expectations_extractor.set_dataset_info`.
+
+## Usage
+
+To begin collecting Airflow DAG metadata with Marquez, use:
+
+```diff
+- from airflow import DAG
++ from marquez_airflow import DAG
```
Comment (wslulciuc) [inline integration/airflow/README.md:94]: Mind replacing occurrences of _marquez_ in the `README.md`?

File: integration/airflow/README.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+1. On DAG **start**, collect metadata for each task using an `Extractor` (the library defines a _default_ extractor to use otherwise)
+2. Collect task input / output metadata (`source`, `schema`, etc)
+3. Collect task run-level metadata (execution time, state, parameters, etc)
+4. On DAG **complete**, also mark the task as _complete_ in Marquez
+
+To enable logging, set the environment variable `MARQUEZ_LOG_LEVEL` to `DEBUG`, `INFO`, or `ERROR`:
+
+```
+$ export MARQUEZ_LOG_LEVEL=INFO
```
Comment (wslulciuc) [inline integration/airflow/README.md:107]: Is `MARQUEZ_LOG_LEVEL` used by the underlying py client for OpenLineage?

File: integration/airflow/README.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+the generated run id for that task. For example, a Spark job can be triggered using the
+`DataProcPySparkOperator` with the correct parent run id using the following configuration:
+```python
+t1 = DataProcPySparkOperator(
+    task_id=job_name,
+    #required pyspark configuration,
+    job_name=job_name,
+    dataproc_pyspark_properties={
+        'spark.driver.extraJavaOptions':
+            f"-javaagent:{jar}={os.environ.get('MARQUEZ_URL')}/api/v1/namespaces/{os.getenv('MARQUEZ_NAMESPACE', 'default')}/jobs/{job_name}/runs/{{{{lineage_run_id(run_id, task)}}}}?api_key={os.environ.get('MARQUEZ_API_KEY')}"
```
Comment (wslulciuc) [inline integration/airflow/README.md:128]: I think we should only document environment variables specific to OpenLineage, though we support Marquez for backwards compatibility.

File: integration/airflow/openlineage/airflow/__init__.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# -*- coding: utf-8 -*-
+
+__author__ = """OpenLineage Project"""
```
Comment (wslulciuc) [inline integration/airflow/openlineage/airflow/__init__.py:15]: The author would just be _OpenLineage_.

File: integration/airflow/setup.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+}
+extras_require["dev"] = set(sum(extras_require.values(), []))
+
+setup(
+    name="openlineage-airflow",
+    version=__version__,
+    description="OpenLineage integration with Airflow",
+    long_description=readme,
+    long_description_content_type="text/markdown",
+    author="OpenLineage Project",
```
Comment (wslulciuc) [inline integration/airflow/setup.py:59]: _Minor_: Any reference to _author_ can just be set to _OpenLineage_.

File: integration/airflow/tests/integration/gcloud/gcloud-service-key.json [hunk 7]
```diff
@@ -0,0 +1 @@
+
```
Comment (wslulciuc) [inline integration/airflow/tests/integration/gcloud/gcloud-service-key.json:1]: We probably wouldn't want to check this file in?

**Review by wslulciuc — CHANGES_REQUESTED** [review]
@mobuchowski Thanks for moving the `marquez-airflow` code over, but can we make sure we also preserve the commit history similar to the PR for Spark #96?

**Comment by mobuchowski** [comment]
Continuing work here: https://github.com/OpenLineage/OpenLineage/pull/110

**Pull request closed without merging**